CommonTools/Statistics/interface/SequentialCombinationGenerator.h

```diff
@@ -99,7 +99,7 @@ template<class T> class SequentialCombinationGenerator {
       }
       int mincnew2=cnew2[0];
       if (ss[n1-1]<mincnew2) return empty;
-      Vecint::iterator j1=find_if(ss.begin(),ss.end(),bind2nd(std::greater<int>(),mincnew2));
+      Vecint::iterator j1=find_if(ss.begin(),ss.end(), [&mincnew2](auto c) { return c > mincnew2;});
```

Comment:
default capture is supposed to capture only the used variables.
So, there is no need to specify the variable itself

Reply:
@guitargeek suggested this change. It works so why not keep it.

Reply:
A little example:
```
#include <iostream>
#include <algorithm>

int main() {

    int x = 5;
    int n = 7;

    auto doStuffAndincrByN = [n](int x) -> int {

        /*
         * ...do stuff in many lines...
         */

        int y = x + n;

        /*
         * ...do stuff in many lines...
         */
        
        return y;
    };

    std::cout << doStuffAndincrByN(x) << std::endl;
}
```
Imagine you are reading the code and want to know by which `n` the lambda function actually increments. With the default capture (`[=]` or `[&]`), you'd have to investigate the body of the lambda function to identify `n`. Even worse, you can't be sure that the do stuff function would have no side effects on the other local variables, in the case of default capture by reference! That's why I don't like `[&]`, because it makes the lambda function impure.

Reply:
OK, it's somewhat generically in the book by S. Meyers.
In google C++ style ``` Use lambda expressions where appropriate. Prefer explicit captures when the lambda will escape the current scope.```
... ```Use default capture by reference ([&]) only when the lifetime of the lambda is obviously shorter than any potential captures.  ```

Here it is apparently a clear case of unnecessary redundancy in the code by using the explicit capture.